app/controllers/formbuilder/edit/code.js

```diff
@@ -34,17 +40,28 @@ export default class FormbuilderEditCodeController extends Controller {
   handleCodeChange = restartableTask(async (newCode) => {
     await timeout(INPUT_DEBOUNCE_MS);
 
-    if (this.formCodeManager.isTtlTheSameAsLatest(newCode)) {
-      return;
-    }
-
-    this.consoleValidateCode(newCode);
+    await this.consoleValidateCode(newCode);
 
     // The newCode is not assigned to this.fromCode as than the editor
     // loses focus as you are updating the content in the editor.
     // Keeping the changes in another variable and at the end assigning
     // the formCode to the updated code
     this.formCodeUpdates = newCode;
+    if (this.ttlHasErrors()) {
+      this.consoleMessages.pushObject({
+        severity: this.getConsoleSeverity(SHACL_SEVERITY_TYPE.info),
+        subject: this.intl.t('messages.subjects.preview'),
+        content: this.intl.t(
+          'messages.feedback.previewNotUpdatedBecauseOfErrorsInTtl'
+        ),
+      });
+      return;
+    }
```

Comment:
When there are errors the ttl code is not saved and put in the preview. This makes it that the application is protected for the shapes we have now. Ofcourse there are other ways to write invalid ttl code but this is a start where we can look at more and better shacl shapes.

Possible todo:
 * new/better shapes
 * checking if the predicates are valid
 * checking if the given object values are referenced in the ttl or of the expected type

This is best implemented in another PR so this one has one issue solved

Reply:
Yes, totally agree on this.